### PR TITLE
feat: add configurable create timeout for VM hosts

### PR DESCRIPTION
**Pull request by SK1Y101**
VM Host operations can take longer than the alloted 20 minutes, this should introduce a user configurable `timeouts` block to increase that time if needed.

### CODE REVIEW

File: maas/resource_maas_instance.go
Comment (skatsaounis) [inline maas/resource_maas_instance.go:208]: I would suggest to revert this change. Being more specific, what this line does is giving the option to our users to explicitly set a timeout when creating `maas_instance` resources. It is also setting the default `create` timeout in case users do not want to set something custom.

If `Timeouts` field is not declared in a resource, Terraform will use the default timeouts for each operations: https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_data.go#L412-L445. As you can see, SDK default choice is also 20 minutes, which I find good.

My suggestion is to bump the timeouts on the resource declarations in the tests that are failing. This can be done with something like this:

```tf
resource "maas_instance" "tf_test_vm_instance" {
  allocate_params {
    hostname = maas_vm_host_machine.tf_test_vm.hostname
  }
  deploy_params {
    distro_series = var.distro_series
  }
  timeouts {
    create = "40m"
  }
}
```

But what we really miss is the ability to declare custom create timeouts on `maas_vm_host` resources. This is extremely useful when we need MAAS to deploy a machine and use it as a VM Host. So I would love to add this with this PR. By the way, this is the problematic resource in our internal tests.

I am more than happy to assist you with the review :slightly_smiling_face:

**Review by skatsaounis — CHANGES_REQUESTED** [review]

File: maas/resource_maas_instance.go
Comment (skatsaounis) [reply maas/resource_maas_instance.go:208]: Here is a link from the SDK documentation if you need to explore more the timeouts: https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/retries-and-customizable-timeouts

Comment (SK1Y101) [reply maas/resource_maas_instance.go]: I think I've added to the correct place instead?

Comment (skatsaounis) [reply maas/resource_maas_instance.go]: Your change looks good. Let's wait for a green CI run before merging it :)

**Review by skatsaounis — APPROVED** [review]
LGTM +1

**Comment by skatsaounis** [comment]
@SK1Y101 could you please update the PR description and merge it? Our e2e tests are happy and green :slightly_smiling_face:

**Pull request merged**
